### PR TITLE
Add CPU and GPU examples for DeciLM-7B

### PR DESCRIPTION
**Pull request by Zhangky11**
## Description
In this PR, we add CPU and GPU examples of DeciLM-7B using bigdl-llm's transformer and optimize_model API.
This is the issue page: https://github.com/analytics-zoo/nano/issues/936

### CODE REVIEW

**Review by shane-huang — APPROVED** [review]
Need to update the table in https://github.com/intel-analytics/BigDL/tree/main/python/llm#verified-models.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/deciLM-7b/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Example: Predict Tokens using `generate()` API
+In the example [generate.py](./generate.py), we show a basic use case for a DeciLM-7B model to predict the next N tokens using `generate()` API, with BigDL-LLM INT4 optimizations.
+### 1. Install
+We suggest using conda to manage environment:
+```bash
+conda create -n llm python=3.9
+conda activate llm
+
+pip install --pre --upgrade bigdl-llm[all] # install the latest bigdl-llm nightly build with 'all' option
+pip install transformers==4.35.2 # required by DeciLM-7B
```
Comment (shane-huang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/deciLM-7b/README.md:16]: maybe we should bump to transformers 4.36 as mixtral? @jason-dai

**Comment by Oscilloscope98** [comment]
Please also updated the readme for Windows GPU supports as we have done here: https://github.com/intel-analytics/BigDL/pull/9992

It would be better to have some verification on Windows GPU.

**Comment by Zhangky11** [comment]
> Please also updated the readme for Windows GPU supports as we have done here: #9992
> 
> It would be better to have some verification on Windows GPU.

Sure, I'll do it.

**Comment by Zhangky11** [comment]
I've updated README file and finished verifications on windows GPU.

**Comment by Zhangky11** [comment]
It can work when I set cpu_embedding=True for both HF-transformer API and Pytorch API.

**Pull request merged**